stubs/docutils/docutils/core.pyi

```diff
@@ -1,3 +1,214 @@
+    def __init__(
+        self,
+        reader: Reader[_S] | None = None,
+        parser: Parser | None = None,
+        writer: Writer | None = None,
+        source: Input[_S] | None = None,
+        source_class: type[Input[_S]] = io.FileInput,
+        destination: Output | None = None,
+        destination_class: type[Output] = io.FileOutput,
+        settings: Values | None = None,
+    ) -> None: ...
+    def set_reader(self, reader_name: str, parser: Parser, parser_name: str) -> None: ...
+    def set_writer(self, writer_name: str) -> None: ...
+    def set_components(self, reader_name: str, parser_name: str, writer_name: str) -> None: ...
+    def setup_option_parser(
+        self,
+        usage: Any | None = None,
+        description: Any | None = None,
+        settings_spec: SettingsSpec | None = None,
+        config_section: str | None = None,
+        **defaults,
+    ) -> OptionParser: ...
+    def get_settings(
+        self,
+        usage: Any | None = None,
+        description: Any | None = None,
+        settings_spec: SettingsSpec | None = None,
+        config_section: str | None = None,
+        **defaults,
+    ) -> Values: ...
+    def process_programmatic_settings(self, settings_spec: SettingsSpec, settings_overrides, config_section: str) -> None: ...
+    def process_command_line(
+        self,
+        argv: list[str] | None = None,
+        usage: Any | None = None,
+        description: Any | None = None,
+        settings_spec: SettingsSpec | None = None,
+        config_section: str | None = None,
+        **defaults,
+    ) -> None: ...
+    def set_io(self, source_path: str | None = None, destination_path: str | None = None) -> None: ...
+    def set_source(self, source: Any | None = None, source_path: str | None = None) -> None: ...
+    def set_destination(self, destination: Any | None = None, destination_path: str | None = None) -> None: ...
+    def apply_transforms(self) -> None: ...
+    def publish(
+        self,
+        argv: list[str] | None = None,
+        usage: Any | None = None,
+        description: Any | None = None,
+        settings_spec: SettingsSpec | None = None,
+        settings_overrides: Any | None = None,
+        config_section: str | None = None,
+        enable_exit_status: bool = False,
+    ) -> Any: ...
+    def debugging_dumps(self) -> None: ...
+    def report_Exception(self, error: Exception) -> None: ...
+    def report_SystemMessage(self, error: Exception) -> None: ...
+    def report_UnicodeError(self, error: Exception) -> None: ...
+
+default_usage: str
+default_description: str
+
+def publish_cmdline(
+    reader: Reader[_S] | None = None,
+    reader_name: str = "standalone",
+    parser: Parser | None = None,
+    parser_name: str = "restructuredtext",
+    writer: Writer | None = None,
+    writer_name: str = "pseudoxml",
+    settings: Values | None = None,
+    settings_spec: SettingsSpec | None = None,
+    settings_overrides: Any | None = None,
```

Comment:
Could probably be typed. Docstring says:

> `settings_overrides`: A dictionary containing application-specific settings defaults that override the defaults of other components. [...]

Without looking at the source, this this probably a `dict`, `Mapping`, or one of the protocols from `_typeshed`. It's okay to leave it as `Incomplete` now, though.

Applies to all the functions below as well.